syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -256,8 +256,11 @@ def add(self, shares: dict, other_shares):
                 to the tensor being added to self.
         """
 
-        # if someone passes in a constant... (i.e., x + 3)
-        if not isinstance(other_shares, dict):
+        if isinstance(other_shares, torch.LongTensor) or isinstance(other_shares, torch.IntTensor):
+            # if someone passes a torch tensor
+            other_shares = other_shares.share(*self.child.keys()).child.child
+        elif not isinstance(other_shares, dict):
```

Comment:
Actually there was an error in .share(): we should add the crypto_provider which can be found in self.crypto_provider